webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSInMemory.java

```diff
@@ -0,0 +1,66 @@
+package com.box.l10n.mojito.service.thirdparty;
+
+import com.box.l10n.mojito.entity.Repository;
+import com.box.l10n.mojito.smartling.request.Binding;
+import com.box.l10n.mojito.smartling.request.Bindings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@ConditionalOnProperty(value = "l10n.ThirdPartyTMS.impl", havingValue = "ThirdPartyTMSInMemory", matchIfMissing = true)
+@Component
+public class ThirdPartyTMSInMemory implements ThirdPartyTMS {
+
+    static Logger logger = LoggerFactory.getLogger(ThirdPartyTMSInMemory.class);
+
+    HashMap<String, byte[]> images = new HashMap<>();
+    HashMap<String, HashMap<String, String>> imageToTextUnitMappings = new HashMap<>();
+
+    /**
+     * Storing the uploaded images, that could be useful for testing but at the moment is useless...
+     * TODO remove if we don't need for testing
```

Comment:
@SergeyDemyanenko storing the data might be useful for testing but if not, it can be removed... doesn't hurt for now